src/babel/traversal/path/index.js

```diff
@@ -10,6 +10,23 @@ export default class NodePath {
     this.parent   = parent;
     this.data     = {};
     this.hub      = hub;
+
+    this.shouldSkip         = false;
+    this.shouldStop         = false;
+    this.removed            = false;
```

Comment:
i would like to recommend `undefiend` over `null.

undefined: no value
null: not a value

but both shape the object correctly.

This allows for `obj.key === undefined` checks for

Reply:
I agree with Stefan, `undefined` is probably better choice.

Reply:
@bmeurer does it have better perf than initializing as `null` and comparing against `null` later? or are you suggesting this because of the mentioned different semantics (no value / not a value)

Reply:
There's probably no performance difference, but I think it's easier to maintain long-term if you keep the invariant that `undefined` stands for *no value* as @stefanpenner pointed out above.